File: GeeksForGeeks/Math/Count-Digits-in-a-Number.js
/**
 * Problem: Count Digits in a Number
 * Platform: GeeksForGeeks
 * Difficulty: Easy
 *
 * Approach:
 * Repeatedly divide the number by 10 and count the iterations.
 *
 * Time Complexity: O(log10(n))
 * Space Complexity: O(1)
 */

class Solution {
    countDigits(n) {
        let count = 0;

        while (n > 0) {
            count++;
            n = Math.floor(n / 10);
        }

        return count;
    }
}

File: GeeksForGeeks/Math/Count-Digits-in-a-Number.py
"""
Problem: Count Digits in a Number
Platform: GeeksForGeeks
Difficulty: Easy

Approach:
Repeatedly divide the number by 10 and count how many times
we can do it before it becomes 0.

Time Complexity: O(log10(n))
Space Complexity: O(1)
"""

class Solution:
    def evenlyDivides(self, n):
        count = 0
        temp = n

        while temp > 0:
            count += 1
            temp //= 10

        return count